pkg/kubelet/dockertools/manager.go

```diff
@@ -1362,6 +1363,15 @@ func (dm *DockerManager) killContainer(containerID kubecontainer.ContainerID, co
 	return err
 }
 
+func (dm *DockerManager) generateContainerEvent(containerID kubecontainer.ContainerID, podName, reason, message string) {
+	ref, ok := dm.containerRefManager.GetRef(containerID)
+	if !ok {
+		glog.Warningf("No ref for pod '%q'", podName)
+		return
+	}
+	dm.recorder.Event(ref, api.EventTypeNormal, reason, message)
```

Comment:
Should be api.EventTypeWarning if we emit only on failures. Maybe change the function name as well to make it clearer. `generateFailedHookEvent` ?